### ISSUE TITLE
Include coverage of submodules?

### ISSUE THREAD

**Issue by Flamefire** [Return-To-The-Roots/s25client#1031]
In a recent commit I tried to include the coverage of the submodules. However this does not work as expected:

- on codecov.io it is not shown at all as the files are not in the same repo
- on coveralls it is shown and included in the final value but you can't open the submodule files as they are not in the same repo
- coverage is not "true coverage" as it includes runs from more than the actual tests of those submodules

So we should:
- exclude coverage from submodules again
- add CI to submodules with coverage collection
- add badges to submodules
- opt. include badges of submodules in the main project

@Flow86 What do you think?

**Comment by Flow86** [comment]
I think I would prefer option 3 "add badges to submodules" and 4 "opt. include badges of submodules in the main project" ?

**Comment by Flamefire** [comment]
Alright. Could you enable CI and coverage for all submodules then? And shall we keep coveralls and codecov also for all submodules? And shall we keep both coverage services? I like coveralls slightly more as the diff (especially coverage diff) is better.

**Comment by Flow86** [comment]
we dont need both I think. You wanted to try out which one was better I think?

**Comment by Flamefire** [comment]
Yes. Both have advantages but I think coveralls is better overall
